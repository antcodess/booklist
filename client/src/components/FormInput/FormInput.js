import React from 'react';

import './FormInput.scss';

export const FormInput = ({ ...formInputProps }) => {
  return (
    <div className='form-input-container'>
      <input className='form-input' { ...formInputProps } />
    </div>
  );
};

export default FormInput; 