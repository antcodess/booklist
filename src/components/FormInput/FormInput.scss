.form-input-container {
  
  .form-input {
    background-color: #f6f6f6;
    width: 100%;
    border: none;
    border-radius: 5px;
    padding: 20px 10px;
    display: block;
    margin: 15px 0;
    outline: none;
    font-family: "Cera GR Bold";
    font-size: 0.94rem;

    &::placeholder {
      color: #BCBCBC;
    }
  }
  input[type='password'] {
    letter-spacing: 0.3em;
    
    &::placeholder {
      letter-spacing: 0;
    }
  }
}