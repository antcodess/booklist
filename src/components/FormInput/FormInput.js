import React from 'react';


import './FormInput.scss';

const FormInput = ({ ...formInputProps }) => {
  return (
    <div className='form-input-container'>
      <input className='form-input' { ...formInputProps }/>
    </div>
  );
};

export default FormInput; 