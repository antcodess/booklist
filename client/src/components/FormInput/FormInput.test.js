import React from 'react';
import { shallow } from 'enzyme';
import { FormInput } from './FormInput';

describe('FormInput Component', () => {
  let wrapper;

  beforeEach(() => {

    const mockProps = {
      name:'email',
      type: 'email',
      value: 'example@example.com',
      label: 'Email',
      placeholder: 'Email', 
      required: true 
    }
    wrapper = shallow(<FormInput { ...mockProps } />);
  });

  it('should render FormInput Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});