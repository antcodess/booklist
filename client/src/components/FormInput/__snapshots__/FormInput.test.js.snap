// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component should render FormInput Component 1`] = `ShallowWrapper {}`;
